### PR TITLE
LLM: Support load BaiChuan model family gguf model

### PR DESCRIPTION
**Pull request by hzjane**
## Description
Support BaiChuan model family gguf model

### 1. Why the change?
Support BaiChuan model family gguf model

### 2. User API changes
No

### 3. Summary of the change 
Can load BaiChuan model family gguf model, for example https://huggingface.co/second-state/Baichuan2-7B-Chat-GGUF/tree/main

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/gguf/models/model_implement/baichuan/configuration_baichuan.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
```
Comment (glorysdj) [inline python/llm/src/bigdl/llm/transformers/gguf/models/model_implement/baichuan/configuration_baichuan.py:1]: need to change the comment to bigdl format

File: python/llm/src/bigdl/llm/transformers/gguf/models/model_implement/baichuan/baichuan.zip
Comment (glorysdj) [inline python/llm/src/bigdl/llm/transformers/gguf/models/model_implement/baichuan/baichuan.zip:1]: why we need this?

File: python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md [hunk 2]
```diff
@@ -2,6 +2,17 @@
 In this directory, you will find examples on how to load GGUF model into `bigdl-llm`. For illustration purposes, we utilize the [llama-2-7b-chat.Q4_0.gguf](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main) as a reference LLaMA2 GGUF model.
 >Note: Only LLaMA2 family models are currently supported
 
+## Support Model Family
+model_family refers to the conf['general.architecture'] of the GGUF file
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:6]: You should list the actual models, not the `conf['general.architecture']` here

Comment (hzjane) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:6]: I have list the actual verified models under this line. The model_family is shown that what model_family we support now refer to [here](https://github.com/intel-analytics/BigDL/pull/9685/files#diff-1e0cbab4590ce1ad4f996f05357e881f092dcfc6ec3245ac2cbb22652a1d716eR46)

Comment (jason-dai) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:6]: Then you should list mistral as mistral, not llama

File: python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md [hunk 3]
```diff
@@ -2,6 +2,17 @@
 In this directory, you will find examples on how to load GGUF model into `bigdl-llm`. For illustration purposes, we utilize the [llama-2-7b-chat.Q4_0.gguf](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main) as a reference LLaMA2 GGUF model.
 >Note: Only LLaMA2 family models are currently supported
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:3]: need to change the note

**Comment by jason-dai** [comment]
Please also update the GPU example

**Review by glorysdj — APPROVED** [review]
LGTM

File: python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md [hunk 4]
```diff
@@ -1,6 +1,17 @@
 # Loading GGUF models
-In this directory, you will find examples on how to load GGUF model into `bigdl-llm`. For illustration purposes, we utilize the [llama-2-7b-chat.Q4_0.gguf](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main) as a reference LLaMA2 GGUF model.
->Note: Only LLaMA2 family models are currently supported
+In this directory, you will find examples on how to load GGUF model into `bigdl-llm`.
+
+## Support Model Family
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:4]: Remove this section

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md [hunk 5]
```diff
@@ -1,6 +1,17 @@
 # Loading GGUF models
-In this directory, you will find examples on how to load GGUF model into `bigdl-llm`. For illustration purposes, we utilize the [llama-2-7b-chat.Q4_0.gguf](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main) as a reference LLaMA2 GGUF model.
->Note: Only LLaMA2 family models are currently supported
+In this directory, you will find examples on how to load GGUF model into `bigdl-llm`.
+
+## Support Model Family
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md:4]: Remove this section

Comment (hzjane) [reply python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF/README.md]: Done.

**Pull request merged**